src/nxdoc/nuxeo-server/administration/observability/tracing.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Tracing
+description: Nuxeo is instrumented to expose tracing to various Tracing systems.
+review:
+    date: '2020-06-10'
+    status: ok
+    comment: ''
+labels:
+    - tracing
+    - monitoring
+toc: true
+tree_item_index: 400
+history:
+  -
+---
+The Nuxeo code is instrumented using [Opencensus](https://opencensus.io) to collect traces
+that are published to a tracing system.
+
+A trace tracks the time spent by Nuxeo processing a request. Each trace consists of one or more spans.
+A span is a unit of work and carries key/value tags and annotations with attributes describing the work being done.
+
+Examples of tags:
+![tags]({{file name='jaeger-tags.png'}} ?w=600,border=true)
+
+Examples of annotations (called logs in Jaeger):
+![annotations]({{file name='jaeger-logs.png'}} ?w=600,border=true)
+
+The maximum number of annotations and attributes per span can be configured with the following options:
+- `tracing.attributes.max` the maximum number of attributes in a span, default to 128, maximum is 512
+- `tracing.annotations.max` the maximum number of annotations in a span, default to 128, maximum is 512
+
+The trace context is propagated to scheduled Work or Stream processing.
+As the result the trace shows all related asynchronous processing that are triggered by an HTTP request.
+
+## Supported Tracing Systems
+
+All tracers require the activation of the metric service using:
+`metrics.enabled=true`.
+
+### Jaeger
+You need to adapt the `tracing.jaeger.url` for your setup:
+```
+metrics.enabled=true
+
+tracing.jaeger.enabled=true
+tracing.jaeger.url=http://jaeger:14268/api/traces
+tracing.jaeger.timeout=10s
+tracing.jaeger.service=nuxeo
+```
+
+### Zipkin
+You need to adapt the `tracing.zipkin.url` for your setup:
+```
+metrics.enabled=true
+
+tracing.zipkin.enabled=true
+tracing.zipkin.url=http://zipkin:9411/api/v2/spans
+```
+
+### Datadog
+
+Since Nuxeo 11.2 there is Datadog APM tracer available, you need to adapt the `tracing.datadog.url` for your setup:
+```
+metrics.enabled=true
+
+tracing.datadog.enabled=true
+tracing.datadog.url=http://localhost:8126/v0.3/traces
+```
+
+## Configuring What is Traced
+Which request is traced can be determined by sampling probability using the option `tracing.sampler.probability`.
+A value of `0` means no sampling, and a value of `1` trace all requests.
+
+Now if you want to understand a specific action it is better to use special HTTP request header to activate tracing.
+
+For instance using `curl`, we can add an explicit trace context using the [standard `traceparent`](https://github.com/w3c/trace-context/blob/master/spec/20-http_request_header_format.md) header:
+
+```bash
+curl -XGET -H "traceparent: 00-a0000000000000000000000000000000-0000000000000001-01" "http://localhost:8080/nuxeo/site/api/v1/path/default-domain/workspaces" -u Administrator:Administrator
```

Comment:
I understand that `a0000000000000000000000000000000` is under the control of user running the curl. Also, `-01` is to enable tracing.

What is `00-` and `-0000000000000001` for? Can we change it like the first one? Is there some kind of reserved keywords?

Reply:
improved by putting the w3c spec link for more information after the description.